### PR TITLE
ENH: write and read native coordinates in VTK

### PR DESCRIPTION
**Pull request by volodia99**

### CODE REVIEW

**Comment by neutrinoceros** [comment]
I rewrote the branch history to help bisect the current issue: vtk outputs are corrupted and unreadable. The breaking commit is eaa23222cce4679094e58591b49063c6973c39ae (as one might expect).

I also changed back the reading logic so that
- all "surprise" fields are processed by the while loop that populates `self.data`
- only then, override reconstructed coordinate fields
this is a much more flexible approach with respect to backward and forward compatibility, though I also expect we'll need to update the reading logic in the while loop so that it also accept fields of arbitrary size, which I'll try now.

**Comment by neutrinoceros** [comment]
I think a fundamental issue with our current approach is that we're writing arbitrary fields in the style of the VTK header but *outside* said header. We need to either:
1) change the style
2) actually add these fields to the header

I think option 2) is preferable but I don't know if it's allowed.

File: src/output/vtk.cpp [hunk 1]
```diff
@@ -375,8 +387,8 @@ void Vtk::WriteHeader(IdfxFileHandler fvtk, real time) {
 #elif VTK_FORMAT == VTK_STRUCTURED_GRID
   ssheader << "DATASET STRUCTURED_GRID" << std::endl;
 #endif
-  // fields: geometry, periodicity, time
-  int nfields = 3;
+  // fields: geometry, periodicity, time, 6 NativeCoordinates (x1l, x2l, x3l, x1, x2, x3)
```
Comment (neutrinoceros) [inline src/output/vtk.cpp:390]:
```suggestion
  // fields: geometry, periodicity, time, 6 NativeCoordinates (x1l, x2l, x3l, x1c, x2c, x3c)
```

**Comment by neutrinoceros** [comment]
found a problem downstream: https://github.com/volodia99/nonos/pull/371#issuecomment-2459596390
Specifically, what I'm seeing is that "cell center" arrays have identical data as cell edges (the latter is correct). It may still be an issue with *reading*, but this needs to be clarified.

**Comment by neutrinoceros** [comment]
all clear, the issue was downstream

**Comment by glesur** [comment]
Thanks for this improvement, that seems to fix the issue of reconstructing the cell coordinates from the VTK cartesian coordinates. However, I think there is a regression: the original version of `vtk_io` reconstructed the cell *faces* and stored it in `xl`. Despite the "l" suffix, this was really all the cell faces, i.e. including the rightmost cell face coordinate (in other words, you have one more cell face than cell centres). With a scheme (c representing the cell centres and | the faces):

```
|        c         |          c         |        c         |        c           |
xl(0)   x(0)     xl(1)       x(1)      xl(2)    x(2)      xl(3)    x(3)       xl(4)
```

This new implementation misses that last rightmost face (`xl(4)` on my scheme), since you actually store ``dataBlock::xl`` which is effectively the cell *left face* coordinates. I think we should actually try to keep this.

**Comment by glesur** [comment]
Actually, my previous comment was incorrect, it does include the last rightmost face. So we're good to go.

**Comment by glesur** [comment]
Manually merged as github wouldn't let me do it from their website.

**Pull request merged**